### ISSUE TITLE
Bug in complex reductions in OpenCL backend

### ISSUE THREAD

**Issue by pavanky** [arrayfire/arrayfire#1006]
```
#include <arrayfire.h>
#include <cstdio>
#include <cstdlib>

using namespace af;

int main(int argc, char *argv[])
{
    try {

        array a = randu(1000, 1000, 3, c32);
        cfloat val;
        unsigned idx;
        max<cfloat>(&val, &idx, a);

    } catch (af::exception& e) {
        fprintf(stderr, "%s\n", e.what());
        throw;
    }

    return 0;
}
```

**Comment by pavanky** [comment]
This code results in the OpenCL kernel not compiling

**Comment by shehzan10** [comment]
Fixed by PR #1009
